src/mongo_logger.ts

```diff
@@ -119,61 +189,273 @@ function parseSeverityFromString(s?: string): SeverityLevel | null {
  */
 function resolveLogPath(
   { MONGODB_LOG_PATH }: MongoLoggerEnvOptions,
-  {
-    mongodbLogPath
-  }: {
-    mongodbLogPath?: unknown;
+  { mongodbLogPath }: ClientLogPathOptions
+): MongoDBLogWritable {
+  if (typeof mongodbLogPath === 'string' && /^stderr$/i.test(mongodbLogPath))
+    return createStdLogger(process.stderr);
+  if (typeof mongodbLogPath === 'string' && /^stdout$/i.test(mongodbLogPath))
+    return createStdLogger(process.stdout);
+
+  if (typeof mongodbLogPath === 'object' && typeof mongodbLogPath?.write === 'function') {
+    return mongodbLogPath;
   }
-): Writable {
-  const isValidLogDestinationString = (destination: string) =>
-    ['stdout', 'stderr'].includes(destination.toLowerCase());
-  if (typeof mongodbLogPath === 'string' && isValidLogDestinationString(mongodbLogPath)) {
-    return mongodbLogPath.toLowerCase() === 'stderr' ? process.stderr : process.stdout;
+
+  if (MONGODB_LOG_PATH && /^stderr$/i.test(MONGODB_LOG_PATH)) {
+    return createStdLogger(process.stderr);
+  }
+  if (MONGODB_LOG_PATH && /^stdout$/i.test(MONGODB_LOG_PATH)) {
+    return createStdLogger(process.stdout);
   }
 
-  // TODO(NODE-4813): check for minimal interface instead of instanceof Writable
-  if (typeof mongodbLogPath === 'object' && mongodbLogPath instanceof Writable) {
-    return mongodbLogPath;
+  return createStdLogger(process.stderr);
+}
+
+/** @internal */
+export interface Log extends Record<string, any> {
+  t: Date;
+  c: MongoLoggableComponent;
+  s: SeverityLevel;
+  message?: string;
+}
+
+/** @internal */
+export interface MongoDBLogWritable {
+  write(log: Log): unknown;
+}
+
+function compareSeverity(s0: SeverityLevel, s1: SeverityLevel): 1 | 0 | -1 {
+  const s0Num = SEVERITY_LEVEL_MAP.getNumericSeverityLevel(s0);
+  const s1Num = SEVERITY_LEVEL_MAP.getNumericSeverityLevel(s1);
+
+  return s0Num < s1Num ? -1 : s0Num > s1Num ? 1 : 0;
+}
+
+/** @internal */
+export type LoggableEvent =
+  //| ConnectionPoolMonitoringEvent
+  | CommandStartedEvent
+  | CommandSucceededEvent
+  | CommandFailedEvent
+  | ConnectionPoolCreatedEvent
+  | ConnectionPoolReadyEvent
+  | ConnectionPoolClosedEvent
+  | ConnectionPoolClearedEvent
+  | ConnectionCreatedEvent
+  | ConnectionReadyEvent
+  | ConnectionClosedEvent
+  | ConnectionCheckedInEvent
+  | ConnectionCheckedOutEvent
+  | ConnectionCheckOutStartedEvent
+  | ConnectionCheckOutFailedEvent;
+
+/** @internal */
+export interface LogConvertible extends Record<string, any> {
+  toLog(): Record<string, any>;
+}
+
+/** @internal */
+export type Loggable = LoggableEvent | LogConvertible;
+
+function isLogConvertible(obj: Loggable): obj is LogConvertible {
+  const objAsLogConvertible = obj as LogConvertible;
+  // eslint-disable-next-line no-restricted-syntax
+  return objAsLogConvertible.toLog !== undefined && typeof objAsLogConvertible.toLog === 'function';
+}
+
+function getHostPort(address: string): { host: string; port: number } {
```

Comment:
```ts
  /** @internal */
  toHostPort(): { host: string; port: number } {
    return { host: this.socketPath ?? this.host ?? '', port: this.port ?? 0 };
  }

  // Usage:
  const { host, port } = HostAddress.fromString(address).toHostPort();
```

Just in an effort to consolidate utilities, we can make this a method on HostAddress, it is supposed to be all our abstractions of address string parsing/handling